### PR TITLE
Add examples to API that support a query.

### PR DESCRIPTION
**Pull request by whitingjr**
## Fixes Issue

fixes issue #796 

## Changes proposed

[x] - Add annotation information on API operations
[X] - Add documentation about Queries

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/site/content/en/docs/Concepts/core-concepts/index.md [hunk 1]
```diff
@@ -126,6 +126,7 @@ In Horreum a `Report Configuration` is used for compiling a summary of informati
 ## Report
 A `Report` is an instance of a `Report Configuration`. Creation date and time is used for differentiating each `Report`.
 
+<<<<<<< HEAD
```
Comment (johnaohara) [inline docs/site/content/en/docs/Concepts/core-concepts/index.md:129]: Can you clean this up, looks like a remnant of conflct?

File: docs/site/content/en/docs/Concepts/core-concepts/index.md [hunk 2]
```diff
@@ -137,3 +138,6 @@ Actions only for a particular [`Test`](#test)
 
 ## Action allow list
 `Horreum` will only allow generic HTTP requests to domains that have been pre-configured in Horreum by an Administrator.
+
+## API Filter Query
+`Horreum` API provides query parameters that are `JSONPath` paths that filter operation results.
```
Comment (johnaohara) [inline docs/site/content/en/docs/Concepts/core-concepts/index.md:143]: Can we link to the `APIFilterQueryReference.md` doc so users can navigate to more details?

File: docs/site/content/en/docs/Reference/APIFilterQueryReference.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+title: Horreum API Filter Query Reference
+description: Reference guide for API Filter Query
+date: 2024-03-27
+weight: 3
+---
+
+In Horreum the API provides the ability to filter results. Used when HTTP API response content type is `application-json`. The `API Filter Query` operates on the server side to reduce payload size.
```
Comment (johnaohara) [inline docs/site/content/en/docs/Reference/APIFilterQueryReference.md:8]:
```suggestion
Some APIs provide the ability to filter results when HTTP API response content type is `application-json`. The `API Filter Query` operates on the server side to reduce payload size.
```

File: docs/site/content/en/docs/Reference/APIFilterQueryReference.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+---
+title: Horreum API Filter Query Reference
```
Comment (johnaohara) [inline docs/site/content/en/docs/Reference/APIFilterQueryReference.md:2]:
```suggestion
title: API Filter Queries
```

Comment (johnaohara) [reply docs/site/content/en/docs/Reference/APIFilterQueryReference.md:2]: The Title is used in the menu, "Horreum" and "Reference" are duplicated information in that context

File: docs/site/content/en/docs/Reference/APIFilterQueryReference.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+In Horreum the API provides the ability to filter results. Used when HTTP API response content type is `application-json`. The `API Filter Query` operates on the server side to reduce payload size.
+
+API operations that support Filter Query provide an HTTP request parameter named *query*. The value of the *query* parameter is a `JSONPath` expression. Any syntax errors in the `JSONPath` will cause the operation to fail with details of the  syntax issue.
+
+To use the Filter Query with an operation that returns a [Run](/docs/concepts/core-concepts/#run) data can be done as follows.
+
+```bash
+curl -s  'http://localhost:8080/api/sql/1/queryrun?query=$.*&array=true'  -H 'content-type: application/json' -H 'Authorization: Bearer '$TOKEN
+{"valid":true,"jsonpath":"$.*","errorCode":0,"sqlState":null,"reason":null,"sql":null,"value":"[\"urn:acme:benchmark:0.1\", [{\"test\": \"Foo\", \"duration\": 10, \"requests\": 123}, {\"test\": \"Bar\", \"duration\": 20, \"requests\": 456}], \"defec8eddeadbeafcafebabeb16b00b5\", \"This gets lost by the transformer\"]"}
```
Comment (johnaohara) [inline docs/site/content/en/docs/Reference/APIFilterQueryReference.md:16]: Can the output JSON be formatted so it is split over multiple lines? atm it is a long single line in the rendered doc

File: docs/site/content/en/docs/Reference/APIFilterQueryReference.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+```
+
+The response contains an enclosing JSON object that is common for Filter Query operations. The *value* property contains an array. The elements of the array contain the values of properties in the uploaded [Run](/docs/concepts/core-concepts/#run) JSON. Note the additional HTTP request query parameter named *array*. Setting *array* is necessary to return dynamically sized results.
+
+To filter the above API operation to retrieve only the *results* property of the uploaded [Run](/docs/concepts/core-concepts/#run) object the *query* parameter is defined as `$.results`
+
+```bash
+curl -s  'http://localhost:8080/api/sql/1/queryrun?query=$.results&array=true'  -H 'content-type: application/json' -H 'Authorization: Bearer '$TOKEN
+{"valid":true,"jsonpath":"$.results","errorCode":0,"sqlState":null,"reason":null,"sql":null,"value":"[[{\"test\": \"Foo\", \"duration\": 10, \"requests\": 123}, {\"test\": \"Bar\", \"duration\": 20, \"requests\": 456}]]"}
```
Comment (johnaohara) [inline docs/site/content/en/docs/Reference/APIFilterQueryReference.md:25]: same as above re multi-line json

**Review by johnaohara — CHANGES_REQUESTED** [review]
There are a handful of minor issues to resolve

File: docs/site/content/en/docs/Concepts/core-concepts/index.md (hunk 1 above)
Comment (whitingjr) [reply docs/site/content/en/docs/Concepts/core-concepts/index.md:129]: well spotted

File: docs/site/content/en/docs/Concepts/core-concepts/index.md (hunk 2 above)
Comment (whitingjr) [reply docs/site/content/en/docs/Concepts/core-concepts/index.md:143]: sure

File: docs/site/content/en/docs/Reference/APIFilterQueryReference.md (hunk 5 above)
Comment (whitingjr) [reply docs/site/content/en/docs/Reference/APIFilterQueryReference.md:16]: sure

File: docs/site/content/en/docs/Reference/APIFilterQueryReference.md (hunk 6 above)
Comment (whitingjr) [reply docs/site/content/en/docs/Reference/APIFilterQueryReference.md:25]: sure

File: docs/site/content/en/docs/Reference/APIFilterQueryReference.md (hunk 5 above)
Comment (whitingjr) [reply docs/site/content/en/docs/Reference/APIFilterQueryReference.md]: I expanded the outer block of JSON. Do you want the Inner block also expanded ?

Comment (johnaohara) [reply docs/site/content/en/docs/Reference/APIFilterQueryReference.md]: I think it is ok to leave it as a string, as the output is not a valid JSON node, but a string represenation

**Review by johnaohara — APPROVED** [review]

**Pull request merged**